### PR TITLE
CMCL-1182: missing docs

### PR DESCRIPTION
**Pull request by gaborkb**
### Purpose of this PR
Added docs based on this report: https://yamato-artifactviewer.prd.cds.internal.unity3d.com/759bf7a1-6c2a-454c-a8a8-556226f16069%2Fpvp%2Fupm-ci~%2Fpvp/com.unity.cinemachine.json.

In some places, I changed the access modifier of the function/property that was missing docs. 

### Testing status

- [ ] Added an automated test
- [ ] Passed all automated tests
- [ ] Manually tested 

### Documentation status

- [ ] Updated [CHANGELOG](https://keepachangelog.com/en/1.0.0/)
- [ ] Updated README (if applicable)
- [x] Commented all public classes, properties, and methods
- [ ] Updated user documentation

### CODE REVIEW

File: com.unity.cinemachine/Editor/Utility/InspectorUtility.cs [hunk 1]
```diff
@@ -122,6 +122,12 @@ public static string NicifyClassName(string name)
             return ObjectNames.NicifyVariableName(name);
         }
         
+        /// <summary>
+        /// Remove the "Cinemachine" prefix, then call the standard Unity Nicify,
+        /// and add (Deprecated) to types with Obsolete attributes.
+        /// </summary>
+        /// <param name="type">The type to nicify as a string</param>
+        /// <returns>The nicified name</returns>
```
Comment (glabute) [inline com.unity.cinemachine/Editor/Utility/InspectorUtility.cs:130]: This entire class should internal

File: com.unity.cinemachine/Runtime/Deprecated/CinemachinePathBase.cs [hunk 2]
```diff
@@ -320,9 +320,8 @@ public float FromPathNativeUnits(float pos, PositionUnits units)
                 pos /= length;
             return pos;
         }
-
-        // to add enable toggle
-        public void OnEnable() {}
+        
+        void OnEnable() {} // to add enable toggle
```
Comment (glabute) [inline com.unity.cinemachine/Runtime/Deprecated/CinemachinePathBase.cs:324]: Should be virtual protected, since this is a base class.

**Review by glabute — CHANGES_REQUESTED** [review]

File: com.unity.cinemachine/Runtime/Deprecated/CinemachinePathBase.cs [hunk 3]
```diff
@@ -320,9 +320,8 @@ public float FromPathNativeUnits(float pos, PositionUnits units)
                 pos /= length;
             return pos;
         }
-
-        // to add enable toggle
-        public void OnEnable() {}
+        
+        protected virtual void OnEnable() {} // to add enable toggle
```
Comment (glabute) [inline com.unity.cinemachine/Runtime/Deprecated/CinemachinePathBase.cs:324]: doesn't this need XML doc now?

**Review by glabute — CHANGES_REQUESTED** [review]

File: com.unity.cinemachine/Runtime/Deprecated/CinemachinePathBase.cs (hunk 3 above)
Comment (gaborkb) [reply com.unity.cinemachine/Runtime/Deprecated/CinemachinePathBase.cs]: I don't think it needs it, but I added

**Review by glabute — APPROVED** [review]

**Pull request merged**